spec/features/merchants/items/index_spec.rb

```diff
@@ -92,6 +152,52 @@
         click_link "Create New Item"
         expect(current_path).to eq("/merchants/#{merchant_1.id}/items/new")
       end
+
+    #  User Story 12 - Merchant Items Index: 5 most popular items
+    #   As a merchant
+    #   When I visit my items index page
+    #   Then I see the names of the top 5 most popular items ranked by total revenue generated
+    #   And I see that each item name links to my merchant item show page for that item
+    #   And I see the total revenue generated next to each item name
+
+    #   Notes on Revenue Calculation:
+
+    #   Only invoices with at least one successful transaction should count towards revenue
+    #   Revenue for an invoice should be calculated as the sum of the revenue of all invoice items
+    #   Revenue for an invoice item should be calculated as the invoice item unit price multiplied
+    #   by the quantity (do not use the item unit price)
+
+      it "displays the top 5 most popular items ranked by total revenue for the one merchant" do
+        visit "/merchants/#{merchant_1.id}/items"
+        within ".top-five-items" do
+          expect(page).to have_content("Top 5 Items")
+
+          expect(page).to have_content("#{item_6.name} - $24,000.00 in sales")
+          expect(page).to have_content("#{item_1.name} - $18,000.00 in sales")
+          expect(page).to have_content("#{item_3.name} - $12,000.00 in sales")
+          expect(page).to have_content("#{item_2.name} - $6,000.00 in sales")
+          expect(page).to have_content("#{item_5.name} - $200.00 in sales")
+          expect(page).to_not have_content("#{item_4.name} - $24.00 in sales")
+        end
+      end
+
+      it "displays a link with each name of the top merchants by revenue" do
+        visit "/merchants/#{merchant_1.id}/items"
+
+        within ".top-five-items" do
+          expect(page).to have_link("#{item_1.name}")
+          expect(page).to have_link("#{item_2.name}")
+          expect(page).to have_link("#{item_3.name}")
+          expect(page).to have_link("#{item_5.name}")
+          expect(page).to have_link("#{item_6.name}")
```

Comment:
Same here as the above review, we could use orderly to test the order since your 'it' block describes this as "top merchants by revenue"

Reply:
I rewrote the above test like you suggested and it now utilizes orderly. I tried putting the link in the orderly operation in this second testand I'm not finding a way to do that. I can interpolate just name though, which is done in the test above. I renamed this it block to be "displays each name as a link in the top 5 merchants by revenue" which kinda plays off of the test aboveasserting that the objects are in order. I was going to copy and paste the item test but that seems like overkill to me at this point.